src/main/java/treechopper/common/handler/TreeHandler.java

```diff
@@ -1,231 +1,198 @@
 package treechopper.common.handler;
 
 import net.minecraft.block.Block;
-import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.SaplingBlock;
+import net.minecraft.block.material.MaterialColor;
+import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.item.ItemStack;
-import net.minecraft.util.EnumFacing;
-import net.minecraft.util.EnumHand;
-import net.minecraft.util.NonNullList;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
-import net.minecraft.world.WorldServer;
-import net.minecraftforge.common.util.FakePlayer;
-import net.minecraftforge.common.util.FakePlayerFactory;
-import treechopper.common.config.ConfigurationHandler;
+import net.minecraft.world.server.ServerWorld;
+import treechopper.common.config.Configuration;
 import treechopper.common.tree.Tree;
 
 import java.util.*;
 
+/**
+ * Handles most things related to the brains of the mod. Most of the business logic
+ * is handled within this class.
+ */
 public class TreeHandler {
-
-  private static Map<UUID, Tree> m_Trees = new HashMap<>();
+  private static Map<UUID, Tree> treeMap = new HashMap<>();
   private Tree tree;
 
-  private static <T> T getLastElement(final Iterable<T> elements) {
-    final Iterator<T> itr = elements.iterator();
-    T lastElement = itr.next();
-
-    while (itr.hasNext()) {
-      lastElement = itr.next();
-    }
-
-    return lastElement;
-  }
-
-  public int AnalyzeTree(World world, BlockPos blockPos, EntityPlayer entityPlayer) {
-
+  /**
+   * Get all related metadata about the tree and build a Tree object.
+   *
+   * @param world Minecraft world
+   * @param treePos BlockPos: x,y,z coordinates of the block being destroyed
+   * @param entityPlayer PlayerEntity: the player who is destroying the block
+   * @return Tree
+   */
+  public Tree analyzeTree(World world, BlockPos treePos, PlayerEntity entityPlayer) {
     Queue<BlockPos> queuedBlocks = new LinkedList<>();
-    Set<BlockPos> tmpBlocks = new HashSet<>();
+    Set<BlockPos> tempAdjacentBlocks = new HashSet<>();
     Set<BlockPos> checkedBlocks = new HashSet<>();
+
     BlockPos currentPos;
-    Block logBlock = world.getBlockState(blockPos).getBlock();
+    Block logBlock = world.getBlockState(treePos).getBlock();
     tree = new Tree();
 
-    queuedBlocks.add(blockPos);
-    tree.InsertWood(blockPos);
+    queuedBlocks.add(treePos);
+    tree.insertLog(treePos);
 
     while (!queuedBlocks.isEmpty()) {
 
       currentPos = queuedBlocks.remove();
       checkedBlocks.add(currentPos);
 
-      tmpBlocks.addAll(LookAroundBlock(logBlock, currentPos, world, checkedBlocks));
-      queuedBlocks.addAll(tmpBlocks);
-      checkedBlocks.addAll(tmpBlocks);
-      tmpBlocks.clear();
+      tempAdjacentBlocks.addAll(lookAroundBlock(logBlock, currentPos, world, checkedBlocks));
+      queuedBlocks.addAll(tempAdjacentBlocks);
+      // TODO: smells bad but inf loops
```

Comment:
can you explain

Reply:
I had some trouble in this section of the code, I ran into an infinite loop around adding the temp blocks to the queue but I don't understand what all this section of code is trying to do.  I added a todo so I can go back and investigate, but I tried to leave the main logic alone so I didn't change it.

Reply:
While is running until there are some wooden blocks around (queuedBlocks). Inside of while there is:
`currentPos = queuedBlocks.remove();` removes one block from queue and set it as origin
`checkedBlocks.add(currentPos);` puts origin block into checked blocks (we will not put it into queue anymore)
`tmpBlocks.addAll(LookAroundBlock(logBlock, currentPos, world, checkedBlocks));` checks around of origin block, ignoring checked blocks
`queuedBlocks.addAll(tmpBlocks);` puts these found wooden blocks into queue
`checkedBlocks.addAll(tmpBlocks);` puts these found wooden blocks into checked blocks (we dont need to check them again -> we dont want to put them into queue again)

So we start in origin of breaking. We put this origin into queue. We remove this origin from queue, mark it as checked and we will check around. All found wooden block except checked ones we will put into queue and we will also mark them as checked. Then we take first block from queue (remove it from queue), mark it as checked and we check around this block. All found wooden blocks except checked blocks we put into queue again. And so on.

Reply:
It should not happen that this loop is inf, since we are adding into queue only wooden and not checked blocks. We are removing block from queue one by one until there is not wooden block around. It is basically breadth-first search algorithm.

Reply:
Are you able to reproduce a situation when this loop is inf?

Reply:
Not with this code as it is. I found the infinite loop while I was trying to change it, so not in this code. I will go remove that comment.